crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -389,19 +389,15 @@ def check_repo_tag
     end
 
     def check_key_file
-      expected = @config.fetch("repomd", {})["key_md5"]
+      expected = @config.fetch("repomd", {})["fingerprint"]
       return true if expected.blank?
 
       key_path = repodata_path.join("repomd.xml.key")
       key_path = repodata_media_path.join("repomd.xml.key") unless key_path.file?
 
       if key_path.file?
-        md5 = Digest::MD5.hexdigest(key_path.read)
-        if expected.is_a?(Array)
-          expected.include? md5
-        else
-          md5 == expected
-        end
+        expected == `LC_ALL=C gpg --with-fingerprint #{key_path} | \
```

Comment:
you're removing the support for handling Arrays, imho this might still be needed.